src/resource/solidDataset.ts

```diff
@@ -515,17 +517,55 @@ export async function createContainerAt(
     ...options,
   };
 
-  const response = await config.fetch(url, {
-    method: "PUT",
-    headers: {
-      Accept: "text/turtle",
-      "Content-Type": "text/turtle",
-      "If-None-Match": "*",
-      // This header should not be required to create a Container,
-      // but ESS currently expects it:
-      Link: `<${ldp.BasicContainer}>; rel="type"`,
-    },
-  });
+  let response;
+  let datasetToReturn;
+
+  if (solidDataset === undefined) {
+    response = await config.fetch(url, {
+      method: "PUT",
+      headers: {
+        Accept: "text/turtle",
+        "Content-Type": "text/turtle",
+        "If-None-Match": "*",
+        // This header should not be required to create a Container,
+        // but ESS currently expects it:
+        Link: `<${ldp.BasicContainer}>; rel="type"`,
+      },
+    });
+
+    datasetToReturn = createSolidDataset();
+  } else {
+    const datasetWithChangelog = internal_withChangeLog(solidDataset);
+
+    // If we are creating a new container, then it can't have existed before,
+    // so we can't be doing an 'update' to it. So if the provided dataset has
+    // updates then we assume there was a programmer error. (This may require
+    // changing based on user feedback (e.g., perhaps the dataset was read
+    // from a different container, we augmented that, and now want to use it
+    // as meta-data for our new container)).
+    if (isUpdate(datasetWithChangelog, url)) {
+      throw new Error(
+        `Attempting to create a new container with a dataset that contains updates!`
+      );
+    }
+
+    response = await config.fetch(url, {
+      method: "PUT",
+      body: await triplesToTurtle(
+        toRdfJsQuads(solidDataset).map(getNamedNodesForLocalNodes)
+      ),
+      headers: {
+        Accept: "text/turtle",
+        "Content-Type": "text/turtle",
+        "If-None-Match": "*",
+        // This header should not be required to create a Container,
+        // but ESS currently expects it:
+        Link: `<${ldp.BasicContainer}>; rel="type"`,
+      },
+    });
+
+    datasetToReturn = solidDataset;
+  }
```

Comment:
```suggestion
if (config.solidDataset !== undefined && isUpdate(internal_withChangeLog(solidDataset), url)) {
  // If we are creating a new container, then it can't have existed before,
  // so we can't be doing an 'update' to it. So if the provided dataset has
  // updates then we assume there was a programmer error. (This may require
  // changing based on user feedback (e.g., perhaps the dataset was read
  // from a different container, we augmented that, and now want to use it
  // as meta-data for our new container)).
  throw new Error(
    `Attempting to create a new container with a dataset that contains updates!`
  );
}

const response = await config.fetch(url, {
    method: "PUT",
    body: config.solidDataset 
      ? await triplesToTurtle(toRdfJsQuads(solidDataset).map(getNamedNodesForLocalNodes))
      : undefined,
    headers: {
      Accept: "text/turtle",
      "Content-Type": "text/turtle",
      "If-None-Match": "*",
      // This header should not be required to create a Container,
      // but ESS currently expects it:
      Link: `<${ldp.BasicContainer}>; rel="type"`,
    },
  });
```

Throwing the error early allows to reduce code duplication.